### PR TITLE
chore(date): remove classic theme support - FE-3345

### PR DESCRIPTION
**Pull request by mkrds**
BREAKING CHANGE: Date component no longer supports classic theme

This PR removes classic theme support from experimental Date component, also stories are refactored to use new CSF/MDX formats.

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### CODE REVIEW

**Review by edleeks87 — APPROVED** [review]

**Review by nicktitchmarsh — DISMISSED** [review]

File: cypress/features/regression/experimental/dateInput.feature [hunk 1]
```diff
@@ -3,27 +3,27 @@ Feature: Experimental Date Input component
 
   @positive
   Scenario: Disable Date Input
-    When I open default "Experimental Date Input" component in noIFrame with "dateInput" json from "experimental" using "disabled" object name
+    When I open default "Experimental Date Input Test" component in noIFrame with "dateInput" json from "experimental" using "disabled" object name
```
Comment (DlgSHi) [inline cypress/features/regression/experimental/dateInput.feature:6]: this TC could be deleted - it is covered by `chromatic`

File: cypress/features/regression/experimental/dateInput.feature [hunk 2]
```diff
@@ -3,27 +3,27 @@ Feature: Experimental Date Input component
...
     Then Date input is disabled
 
   @negative
   Scenario: Disable and enable Date Input
-    When I open default "Experimental Date Input" component in noIFrame with "dateInput" json from "experimental" using "disabledFalse" object name
+    When I open default "Experimental Date Input Test" component in noIFrame with "dateInput" json from "experimental" using "disabledFalse" object name
```
Comment (DlgSHi) [inline cypress/features/regression/experimental/dateInput.feature:11]: this TC could be deleted - it is covered by `chromatic`

File: cypress/features/regression/experimental/dateInput.feature [hunk 3]
```diff
@@ -41,12 +41,12 @@ Feature: Experimental Date Input component
 
   @positive
   Scenario: Enable label inline checkbox for Date Input component
-    When I open default "Experimental Date Input" component in noIFrame with "dateInput" json from "experimental" using "labelInline" object name
+    When I open default "Experimental Date Input Test" component in noIFrame with "dateInput" json from "experimental" using "labelInline" object name
```
Comment (DlgSHi) [inline cypress/features/regression/experimental/dateInput.feature:44]: this TC could be deleted - it is covered by `chromatic`

File: cypress/features/regression/experimental/dateInput.feature [hunk 4]
```diff
@@ -41,12 +41,12 @@ Feature: Experimental Date Input component
...
     Then label is inline
 
   @positive
   Scenario Outline: Change Date Input component label align to <labelAlign>
-    When I open default "Experimental Date Input" component in noIFrame with "dateInput" json from "experimental" using "<nameOfObject>" object name
+    When I open default "Experimental Date Input Test" component in noIFrame with "dateInput" json from "experimental" using "<nameOfObject>" object name
```
Comment (DlgSHi) [inline cypress/features/regression/experimental/dateInput.feature:49]: this TC could be converted - and tested by `chromatic`

File: cypress/features/regression/experimental/dateInput.feature [hunk 5]
```diff
@@ -55,7 +55,7 @@ Feature: Experimental Date Input component
 
   @positive
   Scenario Outline: Change Date Input component label width to <width>
-    When I open default "Experimental Date Input" component in noIFrame with "dateInput" json from "experimental" using "<nameOfObject>" object name
+    When I open default "Experimental Date Input Test" component in noIFrame with "dateInput" json from "experimental" using "<nameOfObject>" object name
```
Comment (DlgSHi) [inline cypress/features/regression/experimental/dateInput.feature:58]: this TC could be deleted - it is covered by `chromatic`

File: src/__experimental__/components/date/date.stories.mdx [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+## Quick Start
+
+```jsx
+import DateInput from "carbon-react/lib/__experimental__/components/date";
+```
+
+### Basic
+
+<Preview>
+  <Story name="Basic">
```
Comment (DlgSHi) [inline src/__experimental__/components/date/date.stories.mdx:29]: Could we rename all stories to start with noCapital letter?

**Comment by DlgSHi** [comment]
1. Move `openOnAutoFocus` to the `mdx` file :) 
2. Make changes for the `accessibilityExperimental` for `Date` component

**Review by KatarzynaQA — DISMISSED** [review]

**Review by nicktitchmarsh — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by DlgSHi — DISMISSED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by nicktitchmarsh — DISMISSED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]

**Pull request merged**